pex/testing.py

```diff
@@ -131,29 +148,29 @@ def make_installer(name='my_project',
                    installer_impl=EggInstaller,
                    zip_safe=True,
                    install_reqs=None,
+                   extras_require=None,
                    interpreter=None,
                    **kwargs):
-  interp = {'project_name': name,
-            'version': version,
-            'zip_safe': zip_safe,
-            'install_requires': install_reqs or []}
-  with temporary_content(PROJECT_CONTENT, interp=interp) as td:
+
+  with make_project(name=name,
+                    version=version,
+                    zip_safe=zip_safe,
+                    install_reqs=install_reqs,
+                    extras_require=extras_require) as td:
     yield installer_impl(td, interpreter=interpreter, **kwargs)
 
 
 @contextlib.contextmanager
-def make_source_dir(name='my_project', version='0.0.0', install_reqs=None):
-  interp = {'project_name': name,
-            'version': version,
-            'zip_safe': True,
-            'install_requires': install_reqs or []}
-  with temporary_content(PROJECT_CONTENT, interp=interp) as td:
+def make_source_dir(name='my_project', version='0.0.0', install_reqs=None, extras_require=None):
+  with make_project(name=name,
+                    version=version,
+                    install_reqs=install_reqs,
+                    extras_require=extras_require) as td:
     yield td
 
 
-def make_sdist(name='my_project', version='0.0.0', zip_safe=True, install_reqs=None):
-  with make_installer(name=name, version=version, installer_impl=Packager, zip_safe=zip_safe,
-                      install_reqs=install_reqs) as packager:
+def make_sdist(**kwargs):
```

Comment:
I recommend restoring the original params, in addition to allowing `**kwargs`. It makes the interface simpler to know what parameters it expects.

Reply:
Agreed. Fixed.